### ISSUE TITLE
Add `ripgrep` in documentation to test with `just ci`

### ISSUE THREAD

**Issue by vtamara** [casey/just#2491]
After compiling `just` on OpenBSD/adJ 7.6beta1 and trying to test with `just ci`  I obtained:

```
 % just ci                                                                                                                                           
./bin/forbid                                                                                                                                                    
+ which rg                                                                      
which: rg: Command not found.                                                   
error: Recipe `forbid` failed on line 123 with exit code 2
```

I discovered that `rg` is  `ripgrep`.  On OpenBSD/adJ I installed with `doas pkg_add ripgrep` and after that I could run the tests.

By the way only 3 tests failed:

```
...
failures:                                                                                                                                                       
                                                                                                                                                                
---- attributes::multiple_attributes_one_line stdout ----                                                                                                       
Bad stderr: Diff < left / right > :                                                                                                                             
<error: Justfile contains no recipes.                                                                                                                           
>exit 1                                                                                                                                                         
                                                                                                                                                                    
thread 'attributes::multiple_attributes_one_line' panicked at tests/attributes.rs:59:6:                                                                         
Output mismatch.                                                                                                                                                
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace                                                                                   
                                                                                                                                                                
---- constants::constants_are_not_exported stdout ----                                                                                                          
thread 'constants::constants_are_not_exported' panicked at tests/constants.rs:60:6:                                                                             
Stderr regex mismatch:                                                                                                                                          
"echo $HEXUPPER\nbash: línea 1: HEXUPPER: variable sin asignar\nerror: Recipe `foo` failed on line 4 with exit code 127\n"                                      
!~=                                                                                                                                                             
/Regex("^(?s).*HEXUPPER: unbound variable.*$")/                                                                                                                 
                                                                                                                                                                
---- os_attributes::os stdout ----                                                                                                                              
thread 'os_attributes::os' panicked at tests/os_attributes.rs:59:7:
  unexpected os family                                                                                                                                            
                                                                                                                                                                
                                                                                                                                                                
failures:                                                                                                                                                       
    attributes::multiple_attributes_one_line                                                                                                                    
    constants::constants_are_not_exported                                                                                                                       
    os_attributes::os                                                                                                                                           
                                                                                                                                                                
test result: FAILED. 849 passed; 3 failed; 7 ignored; 0 measured; 0 filtered out; finished in 17.29s                                                            
                                                                                                                                                                
error: test failed, to rerun pass `-p just --test integration`                                                                                                  
error: Recipe `test` failed on line 17 with exit code 101 
```

**Comment by olfa-arfani** [comment]
for the constants_are_not_exported function
would it be possible to replace 
 ```
fn constants_are_not_exported() {
  Test::new()
    .justfile(
      "
        set export

        foo:
          echo $HEXUPPER
      ",
    )
    .stderr_regex(".*HEXUPPER: unbound variable.*")
    .status(127)
    .run();
}
```

by 

 ```
fn constants_are_not_exported() {
  Test::new()
    .justfile(
      "
        foo:
           echo "source $HOME/$HEXUPPER" >> ~/.bashrc
      ",
    )
    .stderr_regex(".*HEXUPPER: unbound variable.*")
    .status(127)
    .run();
}
```
inspired from [https://github.com/casey/just/discussions/1808](https://github.com/casey/just/discussions/1808)

**Comment by laniakea64** [comment]
Please don't make the `just` test suite modify (or risk modifying) customized user configuration files, especially ones that are tricky to get correct and interact with other configurations.

The issue OP is seeing with that test case is that the text "unbound variable" is translated for them, but the test case assumes that part of the error message will always be in English.  Maybe that test should set environment variables to make the shell use `C` locale for error messages?

**Comment by vtamara** [comment]
Thanks @olfa-arfani and @laniakea64 

Running `RUST_BACKTRACE=1 cargo test` I obtain:

```
---- attributes::multiple_attributes_one_line stdout ----                       
Bad stderr: Diff < left / right > :     
<error: Justfile contains no recipes.   
>exit 1                                 
                                        

thread 'attributes::multiple_attributes_one_line' panicked at tests/attributes.rs:59:6:
Output mismatch.                        
stack backtrace:                        
   0: rust_begin_unwind                 
   1: core::panicking::panic_fmt        
   2: integration::test::Test::run      
             at ./tests/test.rs:293:7   
   3: integration::attributes::multiple_attributes_one_line                     
             at ./tests/attributes.rs:48:3                                      
   4: integration::attributes::multiple_attributes_one_line::{{closure}}        
             at ./tests/attributes.rs:47:34                                     
   5: core::ops::function::FnOnce::call_once                                    
             at /usr/obj/ports/rust-1.81.0/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- constants::constants_are_not_exported stdout ----          
thread 'constants::constants_are_not_exported' panicked at tests/constants.rs:60:6:
Stderr regex mismatch:                  
"echo $HEXUPPER\nbash: línea 1: HEXUPPER: variable sin asignar\nerror: Recipe `foo` failed on line 4 with exit code 127\n"
!~=                                     
/Regex("^(?s).*HEXUPPER: unbound variable.*$")/                                 
stack backtrace:                                                                                                                                                
   0: rust_begin_unwind                 
   1: core::panicking::panic_fmt        
   2: integration::test::Test::run      
             at ./tests/test.rs:283:7   
   3: integration::constants::constants_are_not_exported                        
             at ./tests/constants.rs:49:3                                       
   4: integration::constants::constants_are_not_exported::{{closure}}           
             at ./tests/constants.rs:48:32                                      
   5: core::ops::function::FnOnce::call_once                                    
             at /usr/obj/ports/rust-1.81.0/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
                                                                                                                                                                
---- os_attributes::os stdout ----                                                                                                                              
thread 'os_attributes::os' panicked at tests/os_attributes.rs:59:7:             
unexpected os family 
stack backtrace:                                                                                                                                                
   0: rust_begin_unwind                 
   1: core::panicking::panic_fmt                                                                                                                                
   2: integration::os_attributes::os    
             at ./tests/os_attributes.rs:59:7                                   
   3: integration::os_attributes::os::{{closure}}                                                                                                               
             at ./tests/os_attributes.rs:35:8                                   
   4: core::ops::function::FnOnce::call_once                                    
             at /usr/obj/ports/rust-1.81.0/rustc-1.81.0-src/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
                                                                                
                                                                                
failures:                                                                       
    attributes::multiple_attributes_one_line                                    
    constants::constants_are_not_exported                                       
    os_attributes::os                                                                                                                                           
                                                                                                                                                                
test result: FAILED. 849 passed; 3 failed; 7 ignored; 0 measured; 0 filtered out; finished in 21.45s                                                            
                                                                                                                                                                
error: test failed, to rerun pass `--test integration`                          
```

**Comment by casey** [comment]
Thanks for opening this! I just merged #2493, which prints out a better error message in case ripgrep is missing. I think the test failures should be separate issues.
